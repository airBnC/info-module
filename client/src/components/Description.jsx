import PropTypes from 'prop-types';
import React from 'react';

const Description = props => (
  <div id="description" style={props.border}><h1>Description</h1>
    <div id="short-description"><p>{props.short_description}</p></div>
    <div id="more-descriptions">
      {
        props.more_description.map(description => (
          <div key={description.head}><h4>{description.head}</h4>
            <p>{description.body}</p>
          </div>
        ))
      }
    </div>
  </div>);

Description.propTypes = {
  short_description: String,
  more_description: PropTypes.shape([
    {
      head: String,
      body: String,
    },
  ]),
  border: PropTypes.shape({
    border: String,
  }),
};

Description.defaultProps = {
  short_description: 'Short Description',
  more_description: [
    {
      head: 'String',
      body: 'String',
    },
  ],
  border: {
    border: 'solid black 1px',
  },
};

export default Description;
