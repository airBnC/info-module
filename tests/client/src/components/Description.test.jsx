import React from 'react';
import Enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Description from '../../../../client/src/components/Description.jsx';

Enzyme.configure({ adapter: new Adapter() });

const moreDescription = [
  {
    head: 'Mock Head',
    body: 'Mock Body',
  },
];

describe('Description', () => {
  it('should render correctly', () => {
    const output = Enzyme.shallow(<Description
      short_description="Mock Short Description"
      more_description={moreDescription}
    />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
