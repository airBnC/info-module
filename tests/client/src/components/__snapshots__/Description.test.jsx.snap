// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly 1`] = `
<div
  id="description"
  style={
    Object {
      "border": "solid black 1px",
    }
  }
>
  <h1>
    Description
  </h1>
  <div
    id="short-description"
  >
    <p>
      Mock Short Description
    </p>
  </div>
  <div
    id="more-descriptions"
  >
    <div
      key="Mock Head"
    >
      <h4>
        Mock Head
      </h4>
      <p>
        Mock Body
      </p>
    </div>
  </div>
</div>
`;
